packages/slate-react/src/components/slate.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useMemo } from 'react'
+import { Editor, Node, Operation, Range } from 'slate'
+
+import { ReactEditor } from '../plugin/react-editor'
+import { FocusedContext } from '../hooks/use-focused'
+import { EditorContext } from '../hooks/use-editor'
+import { SlateContext } from '../hooks/use-slate'
+import { EDITOR_TO_ON_CHANGE } from '../utils/weak-maps'
+
+/**
+ * A wrapper around the provider to handle `onChange` events, because the editor
+ * is a mutable singleton so it won't ever register as "changed" otherwise.
+ */
+
+export const Slate = (props: {
+  editor: Editor
+  value: Node[]
+  selection: Range | null
+  children: JSX.Element | JSX.Element[]
+  onChange: (children: Node[], selection: Range | null) => void
+  [key: string]: any
+}) => {
+  const { editor, children, onChange, value, selection, ...rest } = props
+  const context: [Editor] = useMemo(() => {
+    editor.children = value
+    editor.selection = selection
+    return [editor]
+  }, [value, selection, ...Object.values(rest)])
+
```

Comment:
i wonder if there will be some gotchas around memoizing around the extra props. for the most part this is probably what a caller is going to need but this seems like a subtle trap for users who are not paying attention

Reply:
Yeah, although by including them it's less restrictive (not more) so I think it's alright? Otherwise implementing your own custom editor properties becomes impossible I think?